src/graph/unit_graph.cc

```diff
@@ -881,9 +1021,11 @@ HeteroGraphPtr UnitGraph::AsNumBits(HeteroGraphPtr g, uint8_t bits) {
     //   be fixed later.
     auto bg = std::dynamic_pointer_cast<UnitGraph>(g);
     CHECK_NOTNULL(bg);
+
     CSRPtr new_incsr = CSRPtr(new CSR(bg->GetInCSR()->AsNumBits(bits)));
     CSRPtr new_outcsr = CSRPtr(new CSR(bg->GetOutCSR()->AsNumBits(bits)));
-    return HeteroGraphPtr(new UnitGraph(g->meta_graph(), new_incsr, new_outcsr, nullptr));
+    return HeteroGraphPtr(
+        new UnitGraph(g->meta_graph(), new_incsr, new_outcsr, nullptr, SparseFormat::ANY));
```

Comment:
This one might worth inheriting the restrict format flag.

Reply:
These two functions are for casting the graph index for message passing, where CSR/CSC must be materialized.  Since they are materialized anyway, the restrict format flag no longer matters.

Reply:
Got it. We'll likely to revisit this when implementing #1120 where AsNumBits and CopyTo would not materialize non-existing formats.